bears/coffee_script/CoffeeLintBear.py

```diff
@@ -22,11 +23,289 @@ class CoffeeLintBear:
+                        no_interpolation_in_single_quotes: bool=True,
+                        no_stand_alone_at: bool=True,
+                        disable_throwing_strings: bool=True,
+                        no_trailing_semicolons: bool=True,
+                        no_trailing_whitespaces: bool=True,
+                        no_trailing_whitespaces_affects_comments: bool=True,
+                        no_trailing_whitespaces_affects_empty_lines: bool=True,
+                        no_unnecessary_double_quotes: bool=False,
+                        use_english_operator: bool=False,
+                        spaces_around_operators: bool=False,
+                        space_after_comma: bool=True,
+                        cyclomatic_complexity: int=10,
+                        prevent_duplicate_keys: bool=True,
+                        consistent_line_endings: bool=False,
+                        consistent_line_endings_style: str='unix',
+                        no_this: bool=False,
+                        no_decr_or_incrementation_operators: bool=False,
+                        no_empty_parameter_list: bool=False,
+                        no_empty_functions: bool=False,
+                        enforce_parentheses_on_non_empty_constructors: bool=True
+                        ):
+        """
+        :param max_line_length:
+            Maximum number of characters per line.
+        :param max_line_length_affects_comments:
+            Determines if ``max_line_length`` should also affects comments or
+            not.
+        :param space_before_and_after_arrow:
+            Determines if spaces should be used before and/or after the arrow.
+        :param check_braces_spacing:
+            Checks if the proper spacing is used inside curly braces.
+        :param braces_spacing_width:
+            Determines the number of blank spaces after the opening ``{`` and
+            before the closing brace ``}`` given that there is something within
+            the braces.
+        :param spacing_in_empty_braces:
+            Determines the number of blank spaces after the opening ``{`` and
+            before the closing brace ``}`` given an empty content.
+        :param camelcase_classes_name:
+            Checks whether the classes name should be in camel-case or not.
+        :param spaces_before_and_after_colon:
+            Determines if blank spaces should be added before and after colons.
+        :param spaces_before_colon:
+            Determines the number of blank spaces before colon when
+            ``spaces_before_and_after_colon`` is `` True``.
+        :param spaces_after_colon:
+            Determines the number of space after colon when
+            ``spaces_before_and_after_colon`` is `` True``.
+        :param check_whether_file_ends_in_newline:
+            Checks if the file ends with a single newline.
+        :param tab_width:
+            Number of spaces for indentation.
```

Comment:
tab_width is number of *spaces* to indent right? spaces exist only if use spaces is true right? otherwise it will use *tabs* and we wont need spaces